libs/libc/netdb/lib_dnsquery.c

```diff
@@ -439,8 +440,11 @@ static int dns_recv_response(int sd, FAR union dns_addr_u *addr, int naddr,
   ninfo("Question: type=%04x, class=%04x\n",
         ntohs(que->type), ntohs(que->class));
 
-  if (que->type  != qinfo->rectype ||
-      que->class != HTONS(DNS_CLASS_IN))
+  /* N.B. Unaligned access may occur here */
+
+  temp = HTONS(DNS_CLASS_IN);
+  if (memcmp(&que->type, &qinfo->rectype, sizeof(uint16_t)) != 0 ||
```

Comment:
But both dns_header_s and dns_question_s are naturally aligned, the unalignment most likely come from buffer. The better fix is ensure buffer variable start at the four bytes boundary.

Reply:
No, DNS request/response struct are packed, changing buffer type does not work.

Reply:
dns_header_s and dns_question_s don't have __attribute__((packed)), why you consider they are packed?

Reply:
dns_question_s is part of the dns response message. There is a variable length area in front of this structure. I am talking about the DNS response message, not the dns_question_s structure.

Reply:
Ok, but line 430-440 need be removed too.

Reply:
You can try to `ping www.baidu.com` and `ping www.baidui.com`, the second domain name will have one more byte, to compare the address of `que`.